clang/test/CodeGenSYCL/free-function-kernel-type-alias-arg.cpp

```diff
@@ -57,7 +64,25 @@ template void bar_using(ns::BarUsing<int>);
 
 // CHECK: template <typename T> void bar_using(ns::Bar<T, float>);
 
+template<typename T1, typename T2>
+[[__sycl_detail__::add_ir_attributes_function("sycl-nd-range-kernel", 2)]]
+void bar_using2(ns::BarUsing2<T1, T2> Arg) {}
+template void bar_using2(ns::BarUsing2<int, float>);
+
+// CHECK: template <typename T1, typename T2> void bar_using2(ns::Bar<ns::Foo<T2>, T1>);
+
 [[__sycl_detail__::add_ir_attributes_function("sycl-nd-range-kernel", 2)]]
 void baz_type(ns::Baz::type Arg) {}
 
 // CHECK: void baz_type(ns::Bar<double, float> Arg);
+
+#if 0
+// This test case fails, but it is added here in advance to add a record of a
```

Comment:
Note for reviewers: this is an additional test case that I came up with, it was not (at least yet) reported by any customer. I do have a work-in-progress patch for it, but it requires bigger changes to address.

I'm adding the known failing test case right now, proceeding with the simpler patch to unblock the reporter. The extra issue that I found will be addressed in a separate PR